File: web/plugins/last_char.js
import { WebPlugin } from "../engine/plugin.js";

export class LastCharPlugin extends WebPlugin {
    constructor() {
        super("文字の末尾");
    }

    async execute(line, context) {
        const match = line.match(/^文字の末尾\s+「(.+)」$/);
        if (!match) return;

        const chars = [...match[1]];

        context.runtime.write(
            chars.length > 0 ? chars[chars.length - 1] : ""
        );
    }
}

File: pc/plugins/last_char.py
import re
from core.plugin import Plugin


class LastCharPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'文字の末尾\s+「(.+)」', line)
        if not match:
            return None

        value = match.group(1)
        return f"print({value!r}[-1] if {value!r} else '')"